### PR TITLE
refactor: keep only one Rest endpoint per resource.

### LINKED ISSUE #26: Refactor to have a single endpoint per entity
Currently we have diferent Jax-rs resources to access the same entity at diferent paths. This is hard to test. One only resource exposes makes thinks easier.

### PR DESCRIPTION
**Pull request by aureamunoz**
fixes #26

### CODE REVIEW

**Review by Sgitario — CHANGES_REQUESTED** [review]
As we're not using the REST Data with Panache extension any longer, we can remove the `ClaimResource` and `ServiceResource`, plus also remove this extension dependency.

File: servicebox-app/src/main/java/io/halkyon/resource/ServiceResource.java [hunk 1]
```diff
@@ -13,5 +13,4 @@
  * The endpoints generated support as HTTP request/response, the `application/json` content type.
  * The endpoints can be overridden, extended using the @see io.halkyon.template.ServicePageResource class.
  */
-@ResourceProperties(path = "services")
 public interface ServiceResource extends PanacheEntityResource<Service, Long> { }
```
[the remark this answers is not in the document]
Comment (aureamunoz) [reply servicebox-app/src/main/java/io/halkyon/resource/ServiceResource.java:16]: We need it for generating Jax-rs resources, don't we? https://quarkus.io/guides/rest-data-panache#hr-generating-resources

File: servicebox-app/src/main/java/io/halkyon/resource/ClaimResource.java [hunk 2]
```diff
@@ -12,5 +12,4 @@
  * The endpoints generated support as HTTP request/response, the `application/json` content type.
  * The endpoints can be overridden, extended using the @see io.halkyon.resource.page.ClaimResource class.
  */
-@ResourceProperties(path = "claims")
 public interface ClaimResource extends PanacheEntityResource<Claim, Long> { }
```
[the remark this answers is not in the document]
Comment (aureamunoz) [reply servicebox-app/src/main/java/io/halkyon/resource/ClaimResource.java:15]: https://quarkus.io/guides/rest-data-panache#hr-generating-resources

**Review by Sgitario — APPROVED** [review]
Ops, sorry. I thought the annotation was mandatory to generate the resources. My bad. 
lgtm

**Review by cmoulliard — COMMENTED** [review]
Yesterday we discussed the idea to use `plural vs singular` form of the REST endpoint (claims -> claim). If we agree to use the plural form, then all the path should be converted to `services`,`/claims` .

**Comment by aureamunoz** [comment]
> Yesterday we discussed the idea to use `plural vs singular` form of the REST endpoint (claims -> claim). If we agree to use the plural form, then all the path should be converted to `services`,`/claims` .

I prefer singular + `/all` path, seems that it makes more sense to me: 
/service/all
/service
/service/byname

WDYT? @Sgitario @cmoulliard

**Comment by Sgitario** [comment]
> > Yesterday we discussed the idea to use `plural vs singular` form of the REST endpoint (claims -> claim). If we agree to use the plural form, then all the path should be converted to `services`,`/claims` .
> 
> I prefer singular + `/all` path, seems that it makes more sense to me: /service/all /service /service/byname
> 
> WDYT? @Sgitario @cmoulliard

I think it's more natural the plural:
- /services -> will retrieve all the services in JSON format
- /services/{id} -> for a single entity and all the POST, PUT, DELETE variants.

**Comment by cmoulliard** [comment]
> * /services -> will retrieve all the services in JSON format

Be aware that for HTMX, content returned is not JSON but HTML ;-)

**Comment by Sgitario** [comment]
> > * /services -> will retrieve all the services in JSON format
> 
> Be aware that for HTMX, content returned is not JSON but HTML ;-)

Yep, I corrected my comment.

**Comment by aureamunoz** [comment]
Yes, I didn't explain enough why the singular made more sense for me. Let me elaborate a little bit more. I agree with you that plural is more natural. The problem here is that we have 3 methods exposed in the same path: 
Captura de Pantalla 2022-10-28 a las 12 48 17

So we need to differentiate those. So we need to custom the path either for 
[the endpoint called from Claims button in home page](https://github.com/aureamunoz/servicebox-poc/blob/63c5794bcc28f9d5873dd5b255de86aa98e858fb/servicebox-app/src/main/java/io/halkyon/resource/page/ClaimResource.java#L35) or [the endpoint called from New Claim button in claims page
](https://github.com/aureamunoz/servicebox-poc/blob/63c5794bcc28f9d5873dd5b255de86aa98e858fb/servicebox-app/src/main/java/io/halkyon/resource/page/ClaimResource.java#L27).

What about `/claims/all` and `/services/all`?

**Comment by Sgitario** [comment]
> What about `/claims/all` and `/services/all`? Wouldn't they be redundant?

Indeed, they are redundant and it needs to be addressed for sure. 

What I did for [the clusters' endpoints](https://github.com/halkyonio/primaza-poc/pull/39) is have these resources:

Captura de pantalla de 2022-10-28 13-05-45
Captura de pantalla de 2022-10-28 13-05-45

Which is the approach I said in this comment: https://github.com/halkyonio/primaza-poc/pull/33#issuecomment-1294774400

**Comment by cmoulliard** [comment]
What about the convention followed by Andy Damevin which is also using singular - https://github.com/ia3andy/renotes/blob/main/src/main/java/rest/Notes.java#L39-L71

```
    @Path("/new-note")
    @Path("/note/{id}")
    @Path("/note/{id}/delete") 
    @Path("/note/{id}/save")
```

**Comment by Sgitario** [comment]
> What about the convention followed by Andy Damevin which is also using singular - https://github.com/ia3andy/renotes/blob/main/src/main/java/rest/Notes.java#L39-L71
> 
> ```
>     @Path("/new-note")
>     @Path("/note/{id}")
>     @Path("/note/{id}/delete") 
>     @Path("/note/{id}/save")
> ```



> What about the convention followed by Andy Damevin which is also using singular - https://github.com/ia3andy/renotes/blob/main/src/main/java/rest/Notes.java#L39-L71
> 
> ```
>     @Path("/new-note")
>     @Path("/note/{id}")
>     @Path("/note/{id}/delete") 
>     @Path("/note/{id}/save")
> ```

And for the list of notes, he is using [this resource](https://github.com/ia3andy/renotes/blob/af19460ba1ab21e5117e23a96a39a1a61c8cede5/src/main/java/rest/Notes.java#L31) which I think it's `@Path("/")`. 

Therefore, if your application it's a notes service, it makes sense these paths. For example: `http://notes-service:8080/` to return a list of notes, or `http://notes-service:8080/note/1` to return the node with ID 1. Though, I think the path `/note` can be omitted. 

Moreover, what about removing the REST Data Panache endpoints (the ones that provide JSON support)? Do we need it?

**Comment by cmoulliard** [comment]
> Moreover, what about removing the REST Data Panache endpoints (the ones that provide JSON support)? Do we need it?

Interesting question as currently we return HTML and not JSON. As the idea of this POC is also to think about the future which is perhaps to develop a backstage plugin, then the door should not be necessarily closed to have the option to return HTML and/or JSON if the frontend is a react app by example - see [here](https://backstage.io/docs/overview/architecture-overview)

**Comment by aureamunoz** [comment]
We could certainly remove the endpoints provided by the extension. I don't like very much the idea to have 2 diferents endpoint to access same resource like I've just commented [here](https://github.com/halkyonio/primaza-poc/pull/39#pullrequestreview-1160028424)

**Comment by Sgitario** [comment]
> We could certainly remove the endpoints provided by the extension. I don't like very much the idea to have 2 diferents endpoint to access same resource like I've just commented [here](https://github.com/halkyonio/primaza-poc/pull/39#pullrequestreview-1160028424)

I agree with @aureamunoz . I don't think the REST Data Panache extension gives an extra value right now and if in the future we need JSON, we can always add the necessary endpoints or re-add the extension. 

About the endpoints, I would be happy to simply have:

- GET: @Path("/services") : to return HTML with the list of services
- POST:@Path("/services") : to consume a form HTML and create a new service
- GET,PUT,DELETE@Path("/services/{id}"): to return or consume HTML with the entity variants

But if you prefer a singular approach:

- GET: @Path("/service/all") : to return HTML with the list of services
- POST:@Path("/service") : to consume a form HTML and create a new service
- GET,PUT,DELETE@Path("/service/{id}"): to return or consume HTML with the entity variants

I don't have a strong opinion about it.

**Comment by cmoulliard** [comment]
> I don't have a strong opinion about it.

Team vote maybe ;-) I like a lot the 2nd one

**Review by cmoulliard — APPROVED** [review]

**Pull request merged**